uv pip install fails due to system python being be used (only if workdir is set).

(Not setting workdir causes issues earlier in the process.)

```
Run uv pip install \
  uv pip install \
    --system \
    -r requirements/ci.txt
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    IMAGE_NAME: maykinmedia/open-archiefbeheer
    DJANGO_SETTINGS_MODULE: openarchiefbeheer.conf.ci
    DOCKER_BUILDKIT: 1
    KEYCLOAK_BASE_URL: http://localhost:28080
    PLAYWRIGHT_SAVE_TRACE: yes
    UV_PYTHON_INSTALL_DIR: /home/runner/work/_temp/uv-python-dir
    UV_PYTHON: 3.12
    UV_CACHE_DIR: /home/runner/work/_temp/setup-uv-cache
Using Python 3.12.3 environment at: /usr
error: The interpreter at /usr is externally managed, and indicates the following:

  To install Python packages system-wide, try apt install
  python3-xyz, where xyz is the package you are trying to
  install.

  If you wish to install a non-Debian-packaged Python package,
  create a virtual environment using python3 -m venv path/to/venv.
  Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
  sure you have python3-full installed.

  If you wish to install a non-Debian packaged Python application,
  it may be easiest to use pipx install xyz, which will manage a
  virtual environment for you. Make sure you have pipx installed.

  See /usr/share/doc/python3.12/README.venv for more information.

hint: Virtual environments were not considered due to the `--system` flag
Error: Process completed with exit code 2.
```

--- a/action.yml
+++ b/action.yml
@@ -89,6 +89,7 @@ runs:
         # Disable cashing on the default branch and tag builds
         enable-cache: "${{ !(github.ref_name == github.event.repository.default_branch || startsWith(github.ref, 'refs/tags/')) }}"
         activate-environment: 'true'
+        prune-cache: 'false'
         cache-dependency-glob: |
           **/requirements/*.txt
           **/requirements/*.in

--- a/action.yml
+++ b/action.yml
@@ -86,7 +86,6 @@ runs:
       id: setup-uv
       with:
         python-version: ${{ inputs.python-version }}
-        working-directory: ${{ inputs.working-directory }}
         # Disable cashing on the default branch and tag builds
         enable-cache: "${{ !(github.ref_name == github.event.repository.default_branch || startsWith(github.ref, 'refs/tags/')) }}"
 

--- a/action.yml
+++ b/action.yml
@@ -88,11 +88,11 @@ runs:
         python-version: ${{ inputs.python-version }}
         # Disable cashing on the default branch and tag builds
         enable-cache: "${{ !(github.ref_name == github.event.repository.default_branch || startsWith(github.ref, 'refs/tags/')) }}"
+        activate-environment: 'true'
 
     - name: Install backend dependencies
       run: |
         uv pip install \
-          --system \
           -r requirements/ci.txt
       shell: bash
       working-directory: ${{ inputs.working-directory }}

--- a/action.yml
+++ b/action.yml
@@ -89,6 +89,16 @@ runs:
         # Disable cashing on the default branch and tag builds
         enable-cache: "${{ !(github.ref_name == github.event.repository.default_branch || startsWith(github.ref, 'refs/tags/')) }}"
         activate-environment: 'true'
+        cache-dependency-glob: |
+          **/requirements/*.txt
+          **/requirements/*.in
+          **/*requirements*.txt
+          **/*requirements*.in
+          **/*constraints*.txt
+          **/*constraints*.in
+          **/pyproject.toml
+          **/uv.lock
+          **/*.py.lock
 
     - name: Install backend dependencies
       run: |

